packages/synapse-react-client/src/components/SetAccessRequirementCommonFields/SetAccessRequirementCommonFields.tsx

```diff
@@ -0,0 +1,327 @@
+      return ACCESS_TYPE.DOWNLOAD
+    case RestrictableObjectType.TEAM:
+      return ACCESS_TYPE.PARTICIPATE
+    default:
+      throw new Error(
+        `RestrictableObjectType ${subject.type} does not have an access type specified.`,
+      )
+  }
+}
+
+export type SetAccessRequirementCommonFieldsHandle = {
+  /* Allow the parent component to trigger saving the AR, so this may be embedded in an arbitrary modal. */
+  save: () => void
+}
+
+export type SetAccessRequirementCommonFieldsProps = {
+  /* Provided when creating a new AR */
+  subject?: RestrictableObjectDescriptor
+  /* Provided when editing an existing AR */
+  accessRequirementId?: string
+  /* Called when AR has been saved successfully */
+  onSave: (
+    accessRequirementId: string,
+    accessRequirementConreteType: ACCESS_REQUIREMENT_CONCRETE_TYPE,
+  ) => void
+  /* Called when error saving AR */
+  onError: () => void
+}
+
+export const SetAccessRequirementCommonFields = React.forwardRef(
+  function SetAccessRequirementCommonFields(
+    props: SetAccessRequirementCommonFieldsProps,
+    ref: React.ForwardedRef<SetAccessRequirementCommonFieldsHandle>,
+  ) {
+    const { subject, accessRequirementId, onSave, onError } = props
+
+    const isEditing = !subject
+
+    const [subjects, setSubjects] = useState<RestrictableObjectDescriptor[]>(
+      subject ? [subject] : [],
+    )
+    const [hasPendingSubjects, setHasPendingSubjects] = useState<boolean>(false)
+    const [subjectsError, setSubjectsError] = useState<string | null>(null)
+    const [name, setName] = useState<string>('')
+    const [nameError, setNameError] = useState<string | null>(null)
+    const [arType, setArType] = useState<ACCESS_REQUIREMENT_CONCRETE_TYPE>(
+      MANAGED_ACT_ACCESS_REQUIREMENT_CONCRETE_TYPE_VALUE,
+    )
+    const [clientError, setClientError] = useState<string | null>(null)
+
+    const onMutationSuccess = (ar: AccessRequirement) => {
+      setClientError(null)
+      onSave(ar.id.toString(), ar.concreteType)
+    }
+
+    const onMutationError = (error: SynapseClientError) => {
+      setClientError(error.reason)
+      onError()
+    }
+
+    const { mutate: createAccessRequirement } = useCreateAccessRequirement({
+      onSuccess: newAr => onMutationSuccess(newAr),
+      onError: error => onMutationError(error),
+    })
+
+    const { mutate: updateAccessRequirement } = useUpdateAccessRequirement({
+      onSuccess: updatedAr => onMutationSuccess(updatedAr),
+      onError: error => onMutationError(error),
+    })
+
+    const {
+      data: accessRequirement,
+      isLoading: isLoadingAr,
+      error: getArError,
+    } = useGetAccessRequirements(accessRequirementId!, {
+      enabled: Boolean(accessRequirementId),
+    })
+
+    useEffect(() => {
+      if (accessRequirement) {
+        setArType(accessRequirement.concreteType)
+        setName(accessRequirement.name)
+        setSubjects(accessRequirement.subjectIds)
+      }
+    }, [accessRequirement])
```

Comment:
When an `accessRequirementId` is provided, then we are editing an existing AR, so fetch that AR and update state accordingly.

Reply:
Opportunity for a useMutation?

Reply:
Since `useGetAccessRequirements` is a GET, `useQuery` might make sense in this case, but agreed I'd like to avoid useQuery + useEffect-> setState if possible, so if there's a better pattern, I'd be happy to change this!